aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -301,7 +301,11 @@ std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool kee
   TORCH_CHECK(self.layout() == Layout::Strided,
               "mode only supports strided layout, got: ", self.layout());
   dim = maybe_wrap_dim(dim, self.dim());
-  if (_dimreduce_return_trivial_no_ident(values, self, dim, keepdim, "mode")) {
+  if (self.numel() == 0) {
+    zero_numel_tensor_resize(values, indices, self, dim, keepdim);
+    return std::forward_as_tuple(values, indices);
```

Comment:
Prefer `std::tie`.
`std::forward` and `std::forward_as_tuple` are used in templated code to achieve perfect forwarding. Even more, as `std::move`, they may break RVO in pre C++17, before "guaranteed copy elision" was introduced.